### PR TITLE
Git pre-push checks for UI

### PR DESCRIPTION
**Pull request by ngotchac**
This would address partially #2291 

A new script in `js/scripts/pre-push` that should be copied to `.git/hooks`.
It runs linting for the js files if the current branch name starts with `js` or `ui`, and if `npm` is installed.
This would prevent this hook to be executed by everybody, but only devs working on the UI.

NB : the linter runs in about 3sec

### CODE REVIEW

File: js/scripts/pre-push [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+branch=$(git rev-parse --abbrev-ref HEAD)
+
+# Check that the branch starts with `ui` or `js`
+# and that `npm` is installed
+if [[ "$branch" =~ ^(ui|js) ]] && npm --version 1>/dev/null 2>/dev/null;
```
Comment (jacogr) [inline js/scripts/pre-push]: I'd rather do it on everything, it is fast enough - plus, we are doing more-and-more of the combined PRs especially in the jsapi realm.

Comment (ngotchac) [inline js/scripts/pre-push]: Okay, agreed. So what about this :

- On `postinstall`, copy the git hook pre-push script to the right folder (like that everyone using npm will have it)
- The script itself only checks if `npm` is installed, as well as `eslint` (globally or in the `node_modules/bin` directory)

File: js/scripts/pre-push [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+base_dir=$(git rev-parse --show-toplevel)
```
Comment (derhuerst) [inline js/scripts/pre-push]: Why not use `npm prefix` here?

Comment (ngotchac) [inline js/scripts/pre-push]: I guess this would be invalidated anyway with `husky`

**Comment by jacogr** [comment]
Looks good. Marked it onice since fist want a proper test (after release) when GitLab is up and running again. Don't want fireworks. So as it stands, looks good, just want to make sure we run all the checks and is prepared for it. (Specifically since we commit from CI)

In short - once GitLab is humming again, can merge, test & firefight if need-be.

**Comment by ngotchac** [comment]
Seems fair. Note that with `husky`, it shouldn't run the hooks on CI

**Comment by gavofyork** [comment]
Here's a UI screenshot
image

**Comment by gavofyork** [comment]
And another 
image

**Comment by gavofyork** [comment]
and another
image

**Comment by gavofyork** [comment]
image

**Comment by gavofyork** [comment]
gitlab seems to be working again

**Comment by jacogr** [comment]
Thanks will get this one in the morning. (Want to make sure we are there to watch - just in-case)

**Pull request merged**